core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -249,4 +257,67 @@ long getMaxNumberOfThreads() {
 
     }
 
+    static class MaxMapCountCheck implements Check {
+
+        private final long limit = 1 << 18;
```

Comment:
> I dont agree, most will be from file based memory maps, which is a fraction of the amount of files. 

This is probably correct, but the point is still that whatever we think the limit on file descriptors should be, the limit on virtual memory areas should be strictly larger because there are other uses of them.

> Like i said, if we do this, we should just go to full mmapdirectory because we lose the benefits of reducing the number of memory maps there: mainly that users don't need to worry about it anymore.

I agree with you but this does not change the fact that the `vm.max_map_count` limit should be strictly greater than the `nofile` limit.